facebookresearch/ParlAI
# Redial refactor; add in option for confounder movies

**Patch description**
Upgrade from `FixedDialogTeacher` -> `DialogTeacher`; start pulling in information from labels, use label for training + custom eval purposes.

**Testing steps**


Output of `parlai eval_model -t redial -m repeat_label -d true` (skipping some lines):
```
...
The Sisterhood of the Traveling Pants                                                                                              
Gods of Egypt
Red
Fame
Escape from Alcatraz
Benji: Off the Leash!
Unmistaken Child                                                                                                                                        
Guardians of the Galaxy Vol. 2
Edward Scissorhands
Jurassic World
Jumanji
Dracula A.D. 1972
Freaky Friday
Dracula: Dead and Loving It
A Bronx Tale
Avatar
Bram Stoker's Dracula
Another 48 Hrs.
Jumanji
Hunter
Hello How are you?
[eval_labels]: What would you like to see? I’m great. Happy new year.
   [RepeatLabelAgent]: What would you like to see? I’m great. Happy new year.
~~
[redial]: I like adventure, action and comedy movies.
[eval_labels]: How about Jumanji  Or the original Jumanji
   [RepeatLabelAgent]: How about Jumanji  Or the original Jumanji
~~
[redial]: Yes, I really want to see that one soon.
[eval_labels]: Me too. Still out in the cinema.
   [RepeatLabelAgent]: Me too. Still out in the cinema.
~~
[redial]: It kind of reminds me of Night at the Museum which I liked!
[eval_labels]: How about Jurassic World Or Guardians of the Galaxy Vol. 2
   [RepeatLabelAgent]: How about Jurassic World Or Guardians of the Galaxy Vol. 2
~~
[redial]: Yes, I liked that one as well. Those are both great selections.
[eval_labels]: Or The Goonies Loved that one.
   [RepeatLabelAgent]: Or The Goonies Loved that one.
~~
[redial]: Everyone likes that movie!
[eval_labels]: Well I hope I helped
   [RepeatLabelAgent]: Well I hope I helped
~~
[redial]: Thanks for all the awesome movie ideas!
[eval_labels]: Your welcome. Bye
   [RepeatLabelAgent]: Your welcome. Bye
...
14:42:18 | Finished evaluating tasks ['redial'] using datatype valid
    accuracy  bleu-4  exs  f1  movie_suggestion_mention
           1   .8957 2157   1                     .9767
(conda_parlai) 
```
